src/Test/Constraints/XmlMatchesXsdConstraint.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Test\Constraints;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ */
+final class XmlMatchesXsdConstraint extends \PHPUnit_Framework_Constraint
+{
+    /**
+     * @var string[]
+     */
+    private $xmlConstraintErrors = array();
+
+    /**
+     * @var string
+     */
+    private $xsd;
+
+    /**
+     * @param string $xsd
+     */
+    public function __construct($xsd)
+    {
+        parent::__construct();
+
+        // replace first only
+        $needle = 'http://www.w3.org/2001/xml.xsd';
+        if (false !== $pos = strpos($xsd, $needle)) {
+            $xsd = substr_replace($xsd, 'file:///'.str_replace('\\', '/', __DIR__).'/xml.xsd', $pos, strlen($needle));
```

Comment:
ok, I must say that I'm super confused about triple `///` and that magical `str_replace` :(

Reply:
The third slash is part of the path (`/C:\foo.txt`), it is required so that the URI can be parsed while omitting the hostname.

Not sure replacing Windows directory separators `\` with regular slashes `/` is required, but it works :)